static/js/controllers/messages-content.js

```diff
@@ -53,8 +53,18 @@ angular.module('inboxControllers').controller('MessagesContentCtrl',
       return LineageModelGenerator.contact(id)
         .catch(function(err) {
           if (err.code === 404) {
-            // might be an unknown contact so do the best we can
-            return { name: id };
+            // We can have messages with no from number, and thus no contact
+
+            // !!!!!!!!Dirty hack!!!!!!!!!!
+            // A 404ing ID could be a phone number, or the ID of a message
+            // with no phone number. If we think it's a phone number return
+            // it as a template so you can reply to it
+            if (id.length >= 32 && id.length <=36) {
+              // Non-existant number
+              return {};
+            } else {
+              return {name: id};
+            }
```

Comment:
Yeah I'm not comfortable with this - it's hard to read and is full of magic numbers that I'm not 100% are right. I think there's another way which is to return `{name: id}` as before. Then when you have the results of the `getContact` and the `MessageContacts` call (line 91 in the updated file), you can look in the `conversation` variable and if the `from` property is empty you can update `name` to `undefined`. It just feels more deterministic...